data_transfer/services/inventory.py

```diff
@@ -9,27 +9,40 @@
 
 
 @lru_cache
-def device_id_by_serial(serial: str) -> Optional[str]:
-    # TODO: there is a rate limit on the inventory!
-    response = requests.get(f"{config.inventory_api}device/byserial/{serial}")
-    # TODO: validation
-    _response = response.json()
-    if not _response["meta"]["success"]:
-        return None
-    return _response["data"]["device_id"]
+def all_devices_by_type(device_type: utils.DeviceType) -> Any:
+    """
+    Retrieve complete list of ALL Devices by model.
+    This is cached as response can be quite large and will be used multiple times per DAG."""
+    model_id = dict(BTF=6, DRM=8)[device_type.name]
+    response = requests.get(f"{config.inventory_api}devices/bytype/{model_id}")
+    return response.json()
```

Comment:
Could do, although at the moment this returns a list of Device objects rather than a list of Serial IDs so we can flexibly use this endpoint as needed.

Shall I change it such that we return only a set of Serial IDs instead?

Reply:
Good point - fair enough! I'd then leave it as is.